File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int max_diag = 0;
        int max_area = 0;
        for (auto dimen : dimensions) {
            int diag = dimen[0]*dimen[0] + dimen[1]*dimen[1];
            int area = dimen[0] * dimen[1];
            if (diag > max_diag) {
                max_diag = diag;
                max_area = area;
            } 
            else if (diag == max_diag) {
                max_area = max(max_area, area); 
            }
        }
        return max_area;
    }
};

File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.py
class Solution:
    def areaOfMaxDiagonal(self, dimensions: List[List[int]]) -> int:
        max_diag,max_area=0,0


        for l,b in dimensions:
            diag=l*l+b*b

            if diag>max_diag:
                max_area=l*b
                max_diag=diag
            
            elif diag==max_diag and max_area<l*b:
                max_area=l*b
        return max_area